File: typescript/src/api/resources/evaluationResults/client/requests/ListResultsEvaluationResultsRequest.ts

import * as Sikaru from "../../../../index.js";

/**
 * @example
 *     {}
 */
export interface ListResultsEvaluationResultsRequest {
    environment?: Sikaru.ListResultsEvaluationResultsRequestEnvironment;
    evaluator?: string | null;
    verdict?: Sikaru.ListResultsEvaluationResultsRequestVerdict | null;
    limit?: number;
    cursor?: string | null;
}

File: ruby/lib/Sikaru/evaluation_results/types/list_results_evaluation_results_request.rb
# frozen_string_literal: true

module Sikaru
  module EvaluationResults
    module Types
      class ListResultsEvaluationResultsRequest < Internal::Types::Model
        field :project_id, -> { String }, optional: false, nullable: false

        field :environment, -> { Sikaru::EvaluationResults::Types::ListResultsEvaluationResultsRequestEnvironment }, optional: true, nullable: false

        field :evaluator, -> { String }, optional: true, nullable: false

        field :verdict, -> { Sikaru::EvaluationResults::Types::ListResultsEvaluationResultsRequestVerdict }, optional: true, nullable: false

        field :limit, -> { Integer }, optional: true, nullable: false

        field :cursor, -> { String }, optional: true, nullable: false
      end
    end
  end
end
